### PR TITLE
Edit message function

### PR DESCRIPTION
**Pull request by howawong**
Connects #92

### CODE REVIEW

**Comment by oursky-travis** [comment]
@howawong, thanks for your PR! By analyzing the history of the files in this pull request, we identified @ben181231 to be a potential reviewer.

File: Dockerfile [hunk 1]
```diff
@@ -1 +1 @@
-FROM quay.io/skygeario/py-skygear:canary-onbuild
+FROM quay.io/skygeario/py-skygear:v0.23.0
```
Comment (rickmak) [inline Dockerfile]: I believe we will want to point to canary-onbuild in the repo.

File: chat/message_history.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+from skygear.options import options as skyoptions
+from skygear.transmitter.encoding import serialize_record
+from skygear.utils.context import current_user_id
+
+
+class MessageHistory:
+    def __init__(self, message):
+        message_record = serialize_record(message.record)
+        self.record = {}
+        for key in ['attachment', 'body', 'metadata',
```
Comment (rickmak) [inline chat/message_history.py]: User `message_record.keys()`?

File: chat/message_handlers.py [hunk 3]
```diff
@@ -146,6 +155,74 @@ def handle_message_after_save(record, original_record, conn):
...
+    SET last_read_message = %(new_last_message_id)s
+    WHERE last_read_message = %(old_last_message_id)s
+    ''', {
+        'schema_name': AsIs(_get_schema_name()),
+        'new_last_message_id': new_last_message_id,
+        'old_last_message_id': last_message.record.id.key
+    })
+
+
+def delete_message(message_id):
```
Comment (rickmak) [inline chat/message_handlers.py]: Implement this in `Message` class?

File: chat/message_history.py (hunk 2 above)
Comment (howawong) [inline chat/message_history.py]: Cannot, because this will also copy other unwanted values such as seq and deleted field

File: Dockerfile (hunk 1 above)
Comment (howawong) [inline Dockerfile]: I got the following message if i use canary-onbuild 

Invalid message: [b'2362-419E', b'', b'{"kind":"event","name":"init","param":{"config":{"http":{"host":":3000"},"app":{"name":"my_skygear_app","api_key":"my_skygear_key","master_key":"my_master_key","access_control":"role","dev_mode":true,"cors_host":"*","slave":false,"response_timeout":60},"database":{"implementation":"pq","option":"postgresql://postgres:@db/postgres?sslmode=disable"},"asset_store":{"implementation":"fs","public":false,"fs":{"url_prefix":"http://localhost:3000/files","secret":""},"s3":{"access_key":"","secret_key":"","region":"","bucket":"","url_prefix":""},"cloud":{"host":"","token":"","public_prefix":"","private_prefix":""}},"apns":{"enable":false,"type":"cert","env":"sandbox","cert_config":{"cert":"","key":""},"token_config":{"team_id":"","key_id":"","key":""}},"gcm":{"enable":false,"api_key":""},"log":{},"zmq":{"timeout":30}}}}'], assuming socket dead

File: chat/message.py [hunk 4]
```diff
@@ -100,6 +101,24 @@ def from_record(cls, record):
...
+        if message is None:
+            raise MessageNotFoundException()
+
+        if message.record['deleted']:
+            raise AlreadyDeletedException()
+
+        message.record['deleted'] = True
+        message.save()
+        return message
+
```
Comment (rickmak) [inline chat/message.py]: Should the delete be instance method as `save`?

I am wondering where we use `save`..? Any clue.

**Pull request merged**